README.md

```diff
@@ -53,10 +53,199 @@ Myget feeds:
+
+### Create basic span
+
+To create the most basic span, you only specify the name. OpenTelemetry SDK collects start/end timestamps, assigns tracing context and assumes status of this span is `OK`.
+
+```csharp
+var span = tracer
+    .SpanBuilder("basic span")
+    .StartSpan();
+
+span.End();
+```
+
+### Create nested spans
+
+In many cases you want to collect nested operations. You can propagate parent spans explicitly in your code or use implicit context propagation embedded into OpenTelemetry and .NET.
+
+#### Explicit parent propagation and assignment
+
+```csharp
+var parentSpan = tracer
+    .SpanBuilder("parent")
+    .StartSpan();
+
+var childSpan = tracer
+    .SpanBuilder("child")
+    .SetParent(parentSpan) // explicitly assigning parent here
+    .StartSpan();
+
+childSpan.End();
+parentSpan.End();
+```
+
+#### Implicit parent propagation and assignment
+
+```csharp
+var parentSpan = tracer
+    .SpanBuilder("parent")
+    .StartSpan();
+
+// calling WithSpan puts parentSpan into the ambient context
+// that flows in async calls.   When child is created, it
+// implicitly becomes child of current span
+using (tracer.WithSpan(parentSpan))
+{
+    var childSpan = tracer
+        .SpanBuilder("child")
+        .StartSpan();
+
+    childSpan.End();
+}
+```
+
+### Span with attributes
+
+Attributes provide additional context on span specific to specific operation it tracks such as HTTP/DB/etc call properties.
+
+```csharp
+var span = tracer
+    .SpanBuilder("span with attributes")
+    // spans have Client, Server, Internal, Producer and Consumer kinds to help visualize them
+    .SetSpanKind(SpanKind.Client)
+    .StartSpan();
+
+// attributes specific to the call
+span.SetAttribute("db.type", "redis");
+span.SetAttribute("db.instance", "localhost:6379[0]");
+span.SetAttribute("db.statement", "SET");
+span.End();
+```
+
+### Span with links
+
+Links allows to create relationships between different traces i.e. allow spans to have multiple relatives. They are typically used to tracer batching scenarios where multiple traces are merged into another one.
+
+Links affect sampling decision and should be added before sampling decision is made (i.e. before span starts).
+
+```csharp
+SpanContext link1 = ExtractContext(eventHubMessage1);
+SpanContext link2 = ExtractContext(eventHubMessage2);
+
+var span = tracer
+    .SpanBuilder("span with links")
+    .SetSpanKind(SpanKind.Server)
+    .AddLink(link1)
+    .AddLink(link2)
+    .StartSpan();
+
+span.End();
+```
+
+### Span with events
+
+Events are timed text (with optional attributes) annotations on the span. Events can be added to current span (or any running span).
+
+```csharp
+var span = tracer
+    .SpanBuilder("incoming HTTP request")
+    .SetSpanKind(SpanKind.Server)
+    .StartSpan();
+
+using (tracer.WithSpan(span))
+{
+    tracer.CurrentSpan.AddEvent("routes reolved");

```

Comment:
example with attributes also.
var attr= new Dictionary<string, object>();
                attr.Add("key1", "val1");
                span.AddEvent("extra  event", attr);

Reply:
I'd like to keep this doc short and not address advanced cases